src/test/java/core/basesyntax/MainTest.java

```diff
@@ -0,0 +1,366 @@
+    private static final String[] EMPTY_FRUIT_NAME = new String[]{"p", "", "10"};
+    private static final String[] EMPTY_OPERATION_TYPE = new String[]{"", "banana", "10"};
+    private static final List<FruitRecord> DATA_BASE = DataBase.DB;
+    private static final Map<String, Integer> FRUIT_COUNT = FruitStorage.FRUIT_COUNT;
+    private static Map<FruitRecord.Type, AmountHandler> strategies;
+    private static List<String> fileData;
+    private static List<FruitRecord> fruitRecords;
+    private static List<String> reportRecords;
+    private static OperationStrategy operationStrategy;
+    private static DataProcessor dataProcessor;
+    private static AmountHandler additionHandler;
+    private static AmountHandler subtractionHandler;
+    private static FileReader fileReaderImpl;
+    private static ReportWriter reportWriter;
+    private static RowParser rowParser;
+    private static DataValidator dataValidator;
+    private static FruitDao fruitDao;
+
+    @Before
+    public void setUp() {
+        additionHandler = new AdditionHandler();
+        subtractionHandler = new SubtractionHandler();
+        fileReaderImpl = new FileReaderImpl();
+        reportWriter = new ReportWriterImpl();
+        rowParser = new RowParserImpl();
+        dataValidator = new DataValidatorImpl();
+        fruitDao = new FruitDaoImpl();
+        fruitRecords = new ArrayList<>();
+        fruitRecords.add(new FruitRecord(20, BALANCE, new Fruit("banana")));
+        fruitRecords.add(new FruitRecord(100, BALANCE, new Fruit("apple")));
+        fruitRecords.add(new FruitRecord(100, SUPPLY, new Fruit("banana")));
+        fruitRecords.add(new FruitRecord(13, PURCHASE, new Fruit("banana")));
+        fruitRecords.add(new FruitRecord(10, RETURN, new Fruit("apple")));
+        fruitRecords.add(new FruitRecord(20, PURCHASE, new Fruit("apple")));
+        fruitRecords.add(new FruitRecord(5, PURCHASE, new Fruit("banana")));
+        fruitRecords.add(new FruitRecord(50, SUPPLY, new Fruit("banana")));
+        reportRecords = new ArrayList<>();
+        reportRecords.add("banana,152");
+        reportRecords.add("apple,90");
+        strategies = new HashMap<>();
+        strategies.put(BALANCE, additionHandler);
+        strategies.put(SUPPLY, additionHandler);
+        strategies.put(RETURN, additionHandler);
+        strategies.put(PURCHASE, subtractionHandler);
+        operationStrategy = new OperationStrategyImpl(strategies);
+        dataProcessor = new DataProcessorImpl();
+        fileData = new ArrayList<>();
+        fileData.add("b,banana,20");
+        fileData.add("b,apple,100");
+        fileData.add("s,banana,100");
+        fileData.add("p,banana,13");
+        fileData.add("r,apple,10");
+        fileData.add("p,apple,20");
+        fileData.add("p,banana,5");
+        fileData.add("s,banana,50");
+    }
+
+    @Test
+    public void getCorrectAmountHandler_OK() {
```

Comment:
this is Operation Strategy Test and you shoud show this in test name

Reply:
Fixed.